tests/test_schema.py

```diff
@@ -246,30 +308,464 @@ class Model(BaseModel):
 
         a: Union[int, str]
         b: List[int]
-        c: Dict[int, Foo]
+        c: Dict[str, Foo]
         d: Union[None, Foo]
         e: Dict[str, Any]
 
-    assert Model.schema() == {
+    model_schema = Model.schema()
+    assert model_schema == {
         'title': 'Model',
         'description': 'party time',
         'type': 'object',
+        'definitions': {
+            'Foo': {
+                'title': 'Foo',
+                'type': 'object',
+                'properties': {'a': {'title': 'A', 'type': 'number'}},
+                'required': ['a'],
+            }
+        },
+        'properties': {
+            'a': {'title': 'A', 'anyOf': [{'type': 'integer'}, {'type': 'string'}]},
+            'b': {'title': 'B', 'type': 'array', 'items': {'type': 'integer'}},
+            'c': {'title': 'C', 'type': 'object', 'additionalProperties': {'$ref': '#/definitions/Foo'}},
+            'd': {'$ref': '#/definitions/Foo'},
+            'e': {'title': 'E', 'type': 'object'},
+        },
+        'required': ['a', 'b', 'c', 'e'],
+    }
+
+
+def test_date_types():
```

Comment:
all these multiple type tests would be better rewritten as parameterised tests - if one case here fails it'll be a lot easier to work out which one an fix if they're 4 separate tests.

Reply:
Done